lib/grover/js/processor.cjs

```diff
@@ -287,7 +284,30 @@ require('readline').createInterface({
   terminal: false,
 }).on('line', function(line) {
   try {
-    Promise.resolve(_processPage.apply(null, JSON.parse(line)))
+
+    let [convertAction, urlOrHtml, options] = JSON.parse(line);
+
+    // Setup imports
+    try {
+      const Module = require('module');
+
+      // resolve puppeteer from the CWD instead of where this script is located
+      try {
+        puppeteer = require(require.resolve('puppeteer', { paths: Module._nodeModulePaths(process.cwd()) }));
+      } catch (e) {
+        if (typeof options.browserWsEndpoint === "string") {
+          puppeteer = require(require.resolve('puppeteer-core', { paths: Module._nodeModulePaths(process.cwd()) }));
```

Comment:
hi @dexterouschen I see you've moved the import into the page processor, but that still doesn't fix the root problem.. and it creates a new problem of the JS script returning to the RB that the code loaded ok when it hasn't loaded yet?! Ie see line 1 of this file. It is returning `ok` when it definitely shouldn't. 

I would see the solution as simply capturing the original load failure, then throwing that when the inner load failure occurs. ie
```
try {
  const Module = require('module');

  try {
    // resolve puppeteer from the CWD instead of where this script is located
    var puppeteer = require(require.resolve('puppeteer', { paths: Module._nodeModulePaths(process.cwd()) }));
  } catch (puppeteerError) {
    try {
      // try resolve `puppeteer-core` library instead
      var puppeteer = require(require.resolve('puppeteer-core', { paths: Module._nodeModulePaths(process.cwd()) }));
    } catch (coreError) {
      // raise the original puppeteer load issue so we don't send people don't the wrong rabbit hole by default.
      throw puppeteerError;
    }
  }
} catch (e) {
  process.stdout.write(JSON.stringify(['err', e.toString()]));
  process.stdout.write("\n");
  process.exit(1);
}

Reply:
@abrom was this part not merged? 
I have
```js
Cannot find module 'puppeteer'. You need to add it to '/app/package.json' and run 'npm install'
```
error because I am using
`puppeteer-core` with a remote docker `alpine-chrome:119` via `browser_ws_endpoint`

```
rover.configure do |config|
  config.options = {
    # executable_path: ENV['PUPPETEER_EXECUTABLE_PATH'],
    launch_args: ['--disable-dev-shm-usage', '--no-sandbox', '--disable-dev-shm-usage'],
    browser_ws_endpoint: ' ws://0.0.0.0:9222/devtools/browser/xxx',
  }
end
```

Reply:
No. The author closed the PR 1 month ago.

Reply:
@abrom just did a PR for that :) https://github.com/Studiosity/grover/pull/225